### ISSUE TITLE
support `--list-variables` in `cluserctl generate cluster` with managed topology clusters

### ISSUE THREAD

**Issue by ykakarap** [kubernetes-sigs/cluster-api#5347]
**User Story**

As a user I would like to be able to run `clusterctl generate cluster --list-variables` even when the template refers to a cluster class. 

**Detailed Description**

The generate cluster command at the moment only deals with one template file from the repository. Therefore listing all the variables in the template is simple. With support for cluster classes in the generate command, the command tends to potentially deal with more than one file. The list of templates to look for variables could be undetermined.

TLDR; the command might need to be able to identify the cluster class template and list all the variables by potentially looking through an unresolved cluster template file.

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue.]

/kind feature

**Comment by fabriziopandini** [comment]
As per 29/7 backlog grooming discussion
/close

**Comment by sbueringer** [comment]
@ykakarap Just for my understanding, what is the current behavior of clusterctl?

With a regular cluster-template it lists all variables (envsubst), with one using ClusterClass it doesn't list any variables?

**Comment by ykakarap** [comment]
The current behavior is that clusterctl will list the variables defined in the cluster templates for both normal and topology based templates. It does not however list any variables that might be in the ClusterClass definition that might be added to the `clusterctl generate cluster` output. 

The issue was originally opened to support also listing variables from the ClusterClass, but now it is closed because as of now we do not support variables in ClusterClass definitions and we do not plan on adding that support any time soon.

**Comment by sbueringer** [comment]
Sounds good. Thx for the context
